src/components/Post/index.tsx

```diff
@@ -51,23 +69,80 @@ const Post: SFC<PostProps> = ({className, post}) => {
     return <S.DropdownMenu icon={mdiDotsVertical} options={menuOptions} />;
   };
 
+  const handleClick = () => {
+    if (!owner.id) return;
+    navigate(`/profile/${owner.id}`);
+  };
+
+  const renderAvatar = () => {
+    if (!owner.id) return <S.Avatar src={owner.avatar} />;
+
+    return (
+      <Link to={`/profile/${owner.id}`}>
+        <S.Avatar src={owner.avatar} />
+      </Link>
+    );
+  };
+
   return (
     <>
       <S.Container className={className}>
         <S.Top>
-          <UserLabel
-            avatar={owner.avatar}
-            description={shortDate(created_date, true)}
-            id={owner.id}
-            username={owner.username}
-          />
+          <S.Text>
+            {renderAvatar()}
+
+            <S.Right>
+              <S.Username $id={owner.id} onClick={handleClick}>
+                {owner.username}
+              </S.Username>
+              <S.Dot>·</S.Dot>
+              <S.Description>{shortDate(created_date, true)}</S.Description>
+            </S.Right>
+          </S.Text>
           {renderDropdownMenu()}
         </S.Top>
         <S.Content>
-          <Linkify>{content}</Linkify>
+          <Linkify>
+            {!showFullContent || content.length <= 400 ? (
+              <>
+                {content}
+                {content.length > 400 && <S.TextLink onClick={toggleShowFullContent}>See less</S.TextLink>}
+              </>
+            ) : (
+              <>
+                {`${content.slice(0, 400)}...`}
+                <S.TextLink onClick={toggleShowFullContent}>See more</S.TextLink>
+              </>
+            )}
+          </Linkify>
         </S.Content>
+
         {image ? <S.Img alt="image" src={image} /> : null}
-        <Comments postId={post.id} />
+        <Line />
+        <S.Div>
+          <S.BoxLeft>
+            <S.Button text="Like" color={ButtonColor.secondary} iconLeft={mdiThumbUpOutline} />
+            <S.Button
+              text="Comment"
+              color={ButtonColor.secondary}
+              iconLeft={mdiComment}
+              iconRight={isOpenCommentBox ? mdiChevronUp : mdiChevronDown}
+              onClick={() => setIsOpenCommentBox(!isOpenCommentBox)}
+              $isOpenCommentBox={isOpenCommentBox}
+            />
+          </S.BoxLeft>
+          <S.BoxRight>
+            <span>22 Likes</span>
+            <S.Dot>·</S.Dot>
+            <span>22 Comments</span>
```

Comment:
These hardcoding also needs to be replaced with actual counts.
We don't have likes at the moment, but we can show the actual counts of comments here